docs/current_docs/manuals/developer/entrypoint-function.mdx

```diff
@@ -0,0 +1,168 @@
+---
+slug: /manuals/developer/entrypoint-function
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Entrypoint Function"
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Entrypoint Function
+
+Every Dagger module has an entrypoint function. The default one is generated automatically and has no arguments.
+
+It's possible to write a custom entrypoint function. The mechanism to do this is SDK-specific.
+
+This is a simple way to accept module-wide configuration, or just to set a few attributes without having to create setter functions for them.
+
+## Simple entrypoint function
+
+The entrypoint function for a module can be overridden by registering a custom constructor. Its parameters are available as flags in the `dagger call` command directly.
+
+:::note
+Dagger modules have only one entrypoint function. Entrypoint function of [custom types](./custom-types.mdx) are not registered; they are constructed by the function that [chains](./chaining.mdx) them.
+:::
+
+Here is an example module with a custom constructor as entrypoint function:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/entrypoint/go/simple/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/entrypoint/python/simple/main.py
+```
+
+:::info
+In the Python SDK, the [`@dagger.object_type`](https://dagger-io.readthedocs.io/en/latest/module.html#dagger.object_type) decorator wraps [`@dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html), which means that an `__init__()` method is automatically generated, with parameters that match the declared class attributes.
+:::
+
+The code listing above is an example of an object that has typed attributes.
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./snippets/entrypoint/typescript/simple/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Here is an example call for this module:
+
+```shell
+dagger call --name=Foo message
+```
+
+The result will be:
+
+```shell
+Hello, Foo!
+```
+
+:::note
+If you plan to use constructor fields in other module functions, ensure that they are declared as public (in Go and TypeScript). This is because Dagger stores fields using serialization and private fields are omitted during the serialization process. As a result, if a field is not declared as public, calling methods that use it will produce unexpected results.
```

Comment:
I understand that we're multi language but here you mention Go, we should have a specific Go section instead?